xarray/tests/test_dataset.py

```diff
@@ -2184,13 +2184,38 @@ def test_resample_and_first(self):
                       'bar': ('time', np.random.randn(10), {'meta': 'data'}),
                       'time': times})
 
-        actual = ds.resample('1D', dim='time', how='first', keep_attrs=True)
+        actual = ds.resample(time='1D').first(keep_attrs=True)
         expected = ds.isel(time=[0, 4, 8])
         self.assertDatasetIdentical(expected, actual)
 
         # upsampling
         expected_time = pd.date_range('2000-01-01', freq='3H', periods=19)
         expected = ds.reindex(time=expected_time)
+        actual = ds.resample(time='3H')
+        for how in ['mean', 'sum', 'first', 'last', ]:
+            method = getattr(actual, how)
+            result = method()
+            self.assertDatasetEqual(expected, result)
```

Comment:
No need to change, but in future you can use pytest parameters for these sorts of loops, and `assert_equal` rather than the `self...` methods throughout